src/asio_server_http2_impl.cc

```diff
@@ -78,6 +78,11 @@ http2_impl::io_services() const {
   return server_->io_services();
 }
 
+int http2_impl::port() const {
+  const auto ports = server_->ports();
+  return ports.empty() ? 0 : server_->ports().back();
```

Comment:
Is there any reason to just return a single port?  If it is enough to return just 1, it is not necessary to make a temporary vector.

Reply:
Inside the asio_server exists a vector for the acceptors, on each acceptor exists a port. In the tests I made the ASIO server creates 2 acceptors inside the vector, the first one have port 0 and the second one have the port I set. I'm not sure if more than two acceptors can be inside the vector in the nghttp ecosystem, so the vector with ports cope well with it.
However, if only one port is in use and the acceptors vector only have two acceptors, where the second one is the one with the open port number, your suggestion is the correct way to do it.

Reply:
If you have two addresses resolved, you have 2 acceptors (e.g., 127.0.0.1 and ::1).
I'm not sure why you get port 0.  In my test, I got the same pot number for 2 acceptors.

Reply:
You are right. I've made an error and that's why I received always zero in the first acceptor. Also I changed the code to always use a vector. Thanks for the feedback!